editor/components/page-attributes/parent.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+import { stringify } from 'querystringify';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { withInstanceId, withAPIData } from '@wordpress/components';
+import { compose } from '@wordpress/element';
+import { buildTermsTree } from '@wordpress/utils';
+import { TermTreeSelect } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentPostId, getEditedPostAttribute, getCurrentPostType } from '../../store/selectors';
+import { editPost } from '../../store/actions';
+
+export function PageAttributesParent( { parent, postId, postType, items, onUpdateParent } ) {
+	const isHierarchical = get( postType, 'data.hierarchical', false );
+	const parentPageLabel = get( postType, 'data.labels.parent_item_colon' );
+	const pageItems = get( items, 'data', [] );
+	if ( ! isHierarchical || ! parentPageLabel || ! pageItems.length ) {
+		return null;
+	}
+
+	const pagesTree = buildTermsTree( pageItems.filter( item => item.id !== postId ).map( ( item ) => ( {
```

Comment:
FYI: `GET /wp/v2/pages` has an `exclude` argument which could be used to exclude the current post ID instead of filtering it within the client. The big downside is that we'd lose the ability to cache/reuse between any other `withAPIData` requests for pages.

https://developer.wordpress.org/rest-api/reference/pages/#list-pages